### PR TITLE
[CUR-1910] SSO login with Canvas

### PR DESCRIPTION
**Pull request by mohsin-tkxel**

### CODE REVIEW

File: src/Controllers/Content.php [hunk 1]
```diff
@@ -86,6 +86,25 @@ public function index()
             $response->send();
         }elseif ( isset($_SESSION['lti']['issuer_client']) ) {
             $custom_email_id = ParamValidate::getKeyInCustomFields($_SESSION, 'person_email_primary');
+            if ( isset($_SESSION['lti_post']['placement']) && $_SESSION['lti_post']['placement'] == "testing_plc" ) {
```
Comment (aqeel-tkxel) [inline src/Controllers/Content.php]: Fix the placement name.
Follow PSR-12 standards.

no spaces after and before the opening/closing braces.

File: src/Controllers/Content.php [hunk 2]
```diff
@@ -86,6 +86,25 @@ public function index()
...
+                $user_data['lti_client_id'] = $_SESSION['lti']['issuer_client'];
+                $user_data['tool_platform'] = ParamValidate::toolPlatformInfo($_SESSION);
+                $user_data['guid'] = ParamValidate::toolPlatformElement($_SESSION,'guid');
+
+                $build_request_data = http_build_query($user_data);
+
+                // encode user information.
+                $lti_user_info = base64_encode($build_request_data);
+                $studio_login_link = CURRIKI_STUDIO_HOST . "/canvas-lti-sso?sso_info=$lti_user_info";
+                var_dump($studio_login_link); die();
```
Comment (aqeel-tkxel) [inline src/Controllers/Content.php:103]: Remove debugging.

File: src/ParamValidate.php [hunk 3]
```diff
@@ -101,5 +101,20 @@ public static function getKeyInCustomFields($session, $key) {
         return ($custom_fields && property_exists($custom_fields, $key)) ? $custom_fields->{$key} : null;
     }
 
+    public static function toolPlatformElement($session,$element='')
```
Comment (aqeel-tkxel) [inline src/ParamValidate.php:104]: Need space after comma

File: src/Controllers/Content.php [hunk 4]
```diff
@@ -86,6 +86,25 @@ public function index()
...
         }elseif ( isset($_SESSION['lti']['issuer_client']) ) {
             $custom_email_id = ParamValidate::getKeyInCustomFields($_SESSION, 'person_email_primary');
+            if ( isset($_SESSION['lti_post']['placement']) && $_SESSION['lti_post']['placement'] == "testing_plc" ) {
+                $user_data = array();
+                $user_data['email'] = $custom_email_id;
+                $user_data['first_name'] = ParamValidate::getKeyInCustomFields($_SESSION, 'person_name_given');
+                $user_data['last_name'] = ParamValidate::getKeyInCustomFields($_SESSION, 'person_name_family');
+                $user_data['lti_client_id'] = $_SESSION['lti']['issuer_client'];
+                $user_data['tool_platform'] = ParamValidate::toolPlatformInfo($_SESSION);
+                $user_data['guid'] = ParamValidate::toolPlatformElement($_SESSION,'guid');
```
Comment (aqeel-tkxel) [inline src/Controllers/Content.php:96]: Need space after comma

File: src/Controllers/Content.php [hunk 5]
```diff
@@ -86,6 +86,25 @@ public function index()
             $response->send();
         }elseif ( isset($_SESSION['lti']['issuer_client']) ) {
             $custom_email_id = ParamValidate::getKeyInCustomFields($_SESSION, 'person_email_primary');
+            if ( isset($_SESSION['lti_post']['placement']) && $_SESSION['lti_post']['placement'] == "testing_plc" ) {
+                $user_data = array();
```
Comment (aqeel-tkxel) [inline src/Controllers/Content.php:90]: Use [] instead of array()

**Review by aqeel-tkxel — CHANGES_REQUESTED** [review]
Changes requested.
Formatting issues and some others.

File: src/Controllers/Content.php [hunk 6]
```diff
@@ -86,6 +86,24 @@ public function index()
             $response->send();
         }elseif ( isset($_SESSION['lti']['issuer_client']) ) {
             $custom_email_id = ParamValidate::getKeyInCustomFields($_SESSION, 'person_email_primary');
+            if (isset($_SESSION['lti_post']['placement']) && $_SESSION['lti_post']['placement'] == 'canvas_sso') {
```
Comment (fahad-curriki) [inline src/Controllers/Content.php:89]: Use "===" operator if possible.

Comment (fahad-curriki) [reply src/Controllers/Content.php:89]: Also do we need to check for "$_SESSION['lti_post']", if it is set as well.

**Review by fahad-curriki — CHANGES_REQUESTED** [review]
Please review requested changes.

File: src/Controllers/Content.php (hunk 6 above)
Comment (mohsin-tkxel) [reply src/Controllers/Content.php:89]: > Also do we need to check for "$_SESSION['lti_post']", if it is set as well.

isset($_SESSION['lti_post']['placement']) will do the same, doesn't it?

**Review by fahad-curriki — APPROVED** [review]
Approved

**Pull request merged**